### PR TITLE
Add ReCaptcha coverage for Contact GraphQL (SwiftOtter's SOP-348)

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)

According to the official documentation, Magento 2.4.7 comes without ReCaptcha coverage for Contact form GraphQL endpoint.

image

This Pull Requests fills this incomprehensible gap.

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

GraphQL call to verify:
```graphql
mutation {
  contactUs(
    input: {
      email: "test@example.com",
      name: "Test User",
      telephone: "1234567890",
      comment: "This is a test comment."
    }
  ) {
    status
  }
}
```

### Contribution checklist (*)
 - [ ] Author has signed the [Adobe CLA](https://opensource.adobe.com/cla.html)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by lbajsarowicz** [comment]
@ihor-sviziev Can you take a look?

**Comment by lbajsarowicz** [comment]
@ihor-sviziev @sivaschenko ?

**Comment by hostep** [comment]
Tagging @nathanjosiah, maybe he'll be able to move this PR forward

File: ReCaptchaContact/Model/WebapiConfigProvider.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    ) {
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getConfigFor(EndpointInterface $endpoint): ?ValidationConfigInterface
+    {
+        if ($endpoint->getServiceMethod() === 'resolve'
+            && $endpoint->getServiceClass() === ContactUs::class) {
```
Comment (ihor-sviziev) [inline ReCaptchaContact/Model/WebapiConfigProvider.php:30]: I see in other places we have a || here. are you sure it should be &&?
Example: https://github.com/magento/security-package/blob/develop/ReCaptchaCustomer/Model/WebapiConfigProvider.php#L65-L69

File: ReCaptchaContact/etc/di.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
+    <type name="Magento\ReCaptchaWebapiApi\Model\CompositeWebapiValidationConfigProvider">
+        <arguments>
+            <argument name="providers" xsi:type="array">
+                <item name="contactUs" xsi:type="object">Magento\ReCaptchaContact\Model\WebapiConfigProvider</item>
```
Comment (ihor-sviziev) [inline ReCaptchaContact/etc/di.xml]: Why not "Contact", as a module name?
```suggestion
                <item name="contact" xsi:type="object">Magento\ReCaptchaContact\Model\WebapiConfigProvider</item>
```

File: ReCaptchaContact/Model/WebapiConfigProvider.php (hunk 1 above)
Comment (lbajsarowicz) [reply ReCaptchaContact/Model/WebapiConfigProvider.php:30]: @ihor-sviziev After the research, I found out that `\Magento\ReCaptchaSendFriend\Model\WebapiConfigProvider::getConfigFor` is the most in-line with Contact form, that's why it's `&&` not `||`

File: ReCaptchaContact/etc/di.xml (hunk 2 above)
Comment (lbajsarowicz) [reply ReCaptchaContact/etc/di.xml]: Introduced change in the codebase.

**Review by ihor-sviziev — APPROVED** [review]

**Pull request closed without merging**